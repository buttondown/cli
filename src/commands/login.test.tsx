import delay from "delay";
import { render } from "ink-testing-library";
import React from "react";
import { beforeEach, expect, test } from "vitest";
import createConfig from "../config.js";
import Login from "./login.js";

beforeEach(() => {
  // Clear any existing config before each test
  const config = createConfig();
  config.clear();
});

test("should display starting status initially", () => {
  const { lastFrame } = render(<Login />);
  expect(lastFrame()).toMatchSnapshot();
});

test("should persist api key", async () => {
  const randomKey = Math.random().toString(36).slice(2, 15);
  const { stdin, lastFrame } = render(<Login />);

  // Wait a bit for the component to be ready
  await delay(50);

  // Type the API key
  stdin.write(randomKey);

  // Wait for the input to be processed
  await delay(50);

  // Submit with Enter
  stdin.write("\r");

  // Wait for the submit to be processed
  await delay(200);

  expect(lastFrame()).toMatchSnapshot();
  const config = createConfig();
  expect(config.get("apiKey")).toBe(randomKey);
});
