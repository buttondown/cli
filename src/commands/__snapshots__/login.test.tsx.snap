// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should display starting status initially 1`] = `
"Please enter your Buttondown API key:

Enter your API key..."
`;

exports[`should persist api key 1`] = `
"âœ“ Successfully configured API key!

To use a different API key, run this command again."
`;
